src/scheduler.rs

```diff
@@ -565,19 +573,28 @@ async fn fetch_idle(
             .context("store_seen_flags_on_imap")?;
     }
 
-    // Fetch the watched folder.
-    connection
-        .fetch_move_delete(ctx, &mut session, &watch_folder, folder_meaning)
-        .await
-        .context("fetch_move_delete")?;
+    if !ctx.should_delete_to_trash().await?
+        || ctx
+            .get_config(Config::ConfiguredTrashFolder)
+            .await?
+            .is_some()
+    {
+        // Fetch the watched folder.
+        connection
+            .fetch_move_delete(ctx, &mut session, &watch_folder, folder_meaning)
+            .await
+            .context("fetch_move_delete")?;
 
-    // Mark expired messages for deletion. Marked messages will be deleted from the server
-    // on the next iteration of `fetch_move_delete`. `delete_expired_imap_messages` is not
-    // called right before `fetch_move_delete` because it is not well optimized and would
-    // otherwise slow down message fetching.
-    delete_expired_imap_messages(ctx)
-        .await
-        .context("delete_expired_imap_messages")?;
+        // Mark expired messages for deletion. Marked messages will be deleted from the server
+        // on the next iteration of `fetch_move_delete`. `delete_expired_imap_messages` is not
+        // called right before `fetch_move_delete` because it is not well optimized and would
+        // otherwise slow down message fetching.
+        delete_expired_imap_messages(ctx)
+            .await
+            .context("delete_expired_imap_messages")?;
+    } else if folder_config == Config::ConfiguredInboxFolder {
```

Comment:
So, after improving `test_trash_multiple_messages()`, it appeared that we should trigger the full folder scan here